src/coordsim/traffic_predictor/lstm_predictor.py

```diff
@@ -0,0 +1,301 @@
+import pandas as pd
+from sklearn.preprocessing import MinMaxScaler
+from keras.models import Sequential, load_model
+from keras.layers import Dense
+from keras.layers import LSTM
+from coordsim.reader import reader
+import numpy as np
+import random
+from pickle import load, dump
+import os
+import argparse
+
+
+class LSTM_Predictor:
+    """
+    Traffic prediction module based on LSTM neural networks
+    NOTICE: Only works with Poisson arrival and trace based scenarios with single ingress node
+    Based on work done here:
+    https://machinelearningmastery.com/time-series-forecasting-long-short-term-memory-network-python/
+    """
+    def __init__(self, trace, params, training_repeats=10, nb_epochs=10,
+                 weights_dir=False, poisson_data=False):
+        """
+        Initiate the class
+        PARAMETERS:
+        ----------
+        trace: list: The traffic trace from the simulator. Will be used to generated training data.
+        params: SimulatorParams: the simulator parameters class
+        training_repeats: int: Number of repetitions for the training process.
+        nb_epochs: int: the number of epochs to train the neural network in each repetition.
+        run_duration: the simulator interface's run duration.
+        weights_dir: path to weights dir. Set to false when training
+        poisson: set to True to generate training data randomly based on Poisson process, not the arrival mean
```

Comment:
If it currently only works with poisson traffic, are we gonna make it later work with deterministic or MMP traffic aswell? Otherwise I don't see any need for this parameter.